### PR TITLE
Buffer turn data on disk to save memory

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

Solve #3992 by not caching data in memory.

## Tasklist
 - [x] Measure slow down: 5% (on Bayern)
 - [x] review
 - [x] adjust for comments

## Requirements / Relations
 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: include/util/vector_view.hpp [hunk 1]
```diff
@@ -175,9 +177,15 @@ template <typename DataT> void swap(vector_view<DataT> &lhs, vector_view<DataT>
 }
 
 template <typename DataT, storage::Ownership Ownership>
+using InternalOrExternalVector =
+    typename std::conditional<Ownership == storage::Ownership::External,
+                              stxxl::vector<DataT>,
```
Comment (oxidase) [inline include/util/vector_view.hpp]: `stxxl::vector` will not allow to use parallelization.

Comment (TheMarex) [inline include/util/vector_view.hpp]: Yes. We need to reinvestigate this when we try to parallelize that loop.

File: include/storage/serialization.hpp [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+    }
+}
+
+template <typename T>
+inline void write(storage::io::FileWriter &writer, const stxxl::vector<T> &vec)
+{
+    writer.WriteOne(vec.size());
+    for (auto idx : util::irange<std::size_t>(0, vec.size()))
+    {
+        writer.WriteOne<T>(vec[idx]);
```
Comment (daniel-j-h) [inline include/storage/serialization.hpp]: I don't understand why we don't provide an I/O overload for everything providing begin/end or size/data.

Comment (danpat) [inline include/storage/serialization.hpp]: We do - `serializeVector()`

**Pull request merged**